### DISCUSSION TITLE
Is it possible to skip tests from the step definition?

### DISCUSSION THREAD

**Discussion by jmoney2341** [cucumber-rs/cucumber#274] [Q&A] [answered]
From the documentation I have seen, it seems like steps can only return a pass or fail status after it is run. Steps are only returning a "skipped" status if the step definition is missing from the list of defined steps.

Is there any way to send a "skipped" status from within the step definition if certain conditions are met?

I know this can be done with a custom runner but was curious if there is already a built in way to do this?

**Comment by ilslv** [comment, marked as answer]
@jmoney2341 yes this can be done with the custom [`Writer`](https://docs.rs/cucumber/0.19.1/cucumber/trait.Writer.html), that will map [`Passed`](https://docs.rs/cucumber/0.19.1/cucumber/event/enum.Step.html#variant.Passed) or [`Failed`](https://docs.rs/cucumber/0.19.1/cucumber/event/enum.Step.html#variant.Failed) events into a [`Skipped`](https://docs.rs/cucumber/0.19.1/cucumber/event/enum.Step.html#variant.Skipped). We have a similar [`FailOnSkipped`](https://docs.rs/cucumber/0.19.1/cucumber/writer/struct.FailOnSkipped.html) [`Writer`](https://docs.rs/cucumber/0.19.1/cucumber/trait.Writer.html) that does the opposite thing: maps [`Skipped`](https://docs.rs/cucumber/0.19.1/cucumber/event/enum.Step.html#variant.Skipped) event into a [`Failed`](https://docs.rs/cucumber/0.19.1/cucumber/event/enum.Step.html#variant.Failed) promoting it into an error.
